client/my-sites/email/titan-set-up-mailbox/titan-set-up-mailbox-form.tsx

```diff
@@ -0,0 +1,127 @@
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useCreateTitanMailboxMutation } from 'calypso/data/emails/use-create-titan-mailbox-mutation';
+import { NewMailBoxList } from 'calypso/my-sites/email/form/mailboxes/components/new-mailbox-list';
+import { EmailProvider } from 'calypso/my-sites/email/form/mailboxes/types';
+import AddEmailAddressesCardPlaceholder from 'calypso/my-sites/email/gsuite-add-users/add-users-placeholder';
+import { emailManagementTitanSetUpThankYou } from 'calypso/my-sites/email/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { TitanMailboxFields } from 'calypso/data/emails/use-create-titan-mailbox-mutation';
+import type { MailboxForm } from 'calypso/my-sites/email/form/mailboxes';
+import type { MailboxOperations } from 'calypso/my-sites/email/form/mailboxes/components/utilities/mailbox-operations';
+
+import './style.scss';
+
+interface TitanSetUpMailboxFormProps {
+	areSiteDomainsLoaded: boolean;
+	selectedDomainName: string;
+}
+
+interface DispatchCompleteSetupClickProps {
+	canContinue: boolean;
+	dispatch: ( action: Record< string, unknown > ) => void;
+	domainName: string;
+	mailboxName: string;
+}
+
+const goToThankYouPage = ( siteSlug: string, selectedDomainName: string, mailboxName: string ) => {
+	const emailAddress = `${ mailboxName }@${ selectedDomainName }`;
+	page( emailManagementTitanSetUpThankYou( siteSlug, selectedDomainName, emailAddress ) );
+};
+
+const dispatchCompleteSetupClick = ( dispatchProps: DispatchCompleteSetupClickProps ) => {
+	const { canContinue, dispatch, domainName, mailboxName } = dispatchProps;
+
+	dispatch(
+		recordTracksEvent(
```

Comment:
I'm curious, are there any rules of thumbs for choosing between `recordTracksEvent` from `calypso/state/analytics/actions` (Redux action creator) vs `recordTracksEvent` from `calypso/lib/analytics/tracks` (push Tracks event "directly")?

Reply:
None that I know of, personally.

Reply:
Hey folks, thanks for the ping! 

We'd generally recommend **going with `lib/analytics` when possible**, because `state/analytics` is simply a wrapper around it, and we'd generally like to get rid of any unnecessary wrappers.

The rationale is simple: tracks code is spread throughout the codebase, and if we don't rely on Redux and all the underlying custom infrastructure, we have one less piece to carry around throughout the codebase.

Now, what likely makes this confusing is that there was a time when we did recommend using the Redux wrapper. However, that was in times when we worked to centralize Redux and the Calypso implementation as "one solution to rule them all". With `react-query ` and `@wordpress/data` this is no longer the case, and the less coupled we are, the better.